### DISCUSSION TITLE
Adding a contact to a specific list

### DISCUSSION THREAD

**Discussion by davidjnevin** [Jericho/StrongGrid#413] [Q&A] [answered]
Hi Jericho,
I hope you're well. StrongGrid is great. Thank you so much. 
Could you possibly tell me how to add a contact to a specific list, I have the listId and am using 

```csharp
var strongGridClient = new Client(Options.ApiKey);
var contactId = await strongGridClient.Contacts.UpsertAsync(email, firstName, lastName);
```

but I don't know how to pass in my listId eg. 12345678-1234-1234-1234-12345678910

Any help would be greatly appreciated

David

**Comment by Jericho** [comment, marked as answer]
Glad you find StrongGrid useful! 

To answer your question: the `UpsertAsync` method accepts a parameters called `listIds` which allows you to add the contact to one or multiple lists. So your code should look something like this:
```csharp
var listIds = new[]
{
     "12345678-1234-1234-1234-12345678910",
    ... feel free to add more ids if you want to add your contact to more than one list ...
};
var strongGridClient = new Client(Options.ApiKey);
var contactId = await strongGridClient.Contacts.UpsertAsync(email, firstName, lastName, listIds: listIds );
```

**Reply by davidjnevin** [reply]
You are an absolute star Jericho, thank you so much!
